src/database/sqlite/sqlite_aclk_alert.c

```diff
@@ -699,371 +854,203 @@ void aclk_process_send_alarm_snapshot(char *node_id, char *claim_id __maybe_unus
-    alarm_log->exec_path = ae->exec ? strdupz(ae_exec(ae)) : strdupz((char *)string2str(host->health.health_default_exec));
-    alarm_log->conf_source = ae->source ? strdupz(ae_source(ae)) : strdupz((char *)"");
-
-    alarm_log->command = strdupz((char *)edit_command);
-
-    alarm_log->duration = (time_t)ae->duration;
-    alarm_log->non_clear_duration = (time_t)ae->non_clear_duration;
-    alarm_log->status = rrdcalc_status_to_proto_enum((RRDCALC_STATUS)ae->new_status);
-    alarm_log->old_status = rrdcalc_status_to_proto_enum((RRDCALC_STATUS)ae->old_status);
-    alarm_log->delay = ae->delay;
-    alarm_log->delay_up_to_timestamp = (time_t)ae->delay_up_to_timestamp;
-    alarm_log->last_repeat = (time_t)ae->last_repeat;
-
-    alarm_log->silenced =
-        ((ae->flags & HEALTH_ENTRY_FLAG_SILENCED) || (ae->recipient && !strncmp(ae_recipient(ae), "silent", 6))) ?
-            1 :
-            0;
+static int calculate_alert_snapshot_entries(nd_uuid_t *host_uuid)
+{
+    int count = 0;
 
-    alarm_log->value_string = strdupz(ae_new_value_string(ae));
-    alarm_log->old_value_string = strdupz(ae_old_value_string(ae));
+    sqlite3_stmt *res = NULL;
 
-    alarm_log->value = (!isnan(ae->new_value)) ? (NETDATA_DOUBLE)ae->new_value : 0;
-    alarm_log->old_value = (!isnan(ae->old_value)) ? (NETDATA_DOUBLE)ae->old_value : 0;
+    if (!PREPARE_STATEMENT(db_meta, SQL_COUNT_SNAPSHOT_ENTRIES, &res))
+        return 0;
 
-    alarm_log->updated = (ae->flags & HEALTH_ENTRY_FLAG_UPDATED) ? 1 : 0;
-    alarm_log->rendered_info = strdupz(ae_info(ae));
-    alarm_log->chart_context = strdupz(ae_chart_context(ae));
-    alarm_log->chart_name = strdupz(ae_chart_name(ae));
+    int param = 0;
+    SQLITE_BIND_FAIL(done, sqlite3_bind_blob(res, ++param, host_uuid, sizeof(*host_uuid), SQLITE_STATIC));
 
-    alarm_log->transition_id = strdupz((char *)transition_id);
-    alarm_log->event_id = (uint64_t) ae->alarm_event_id;
+    param = 0;
+    int rc = sqlite3_step_monitored(res);
+    if (rc == SQLITE_ROW)
+        count = sqlite3_column_int(res, 0);
+    else
+        error_report("Failed to select snapshot count");
 
-    alarm_log->summary = strdupz(ae_summary(ae));
+done:
+    REPORT_BIND_FAIL(res, param);
+    SQLITE_FINALIZE(res);
 
-    freez(edit_command);
+    return count;
 }
-#endif
-
-#ifdef ENABLE_ACLK
-static bool have_recent_alarm_unsafe(RRDHOST *host, int64_t alarm_id, int64_t mark)
-{
-    ALARM_ENTRY *ae = host->health_log.alarms;
 
-    while (ae) {
-        if (ae->alarm_id == alarm_id && ae->unique_id >mark &&
-            (ae->new_status != RRDCALC_STATUS_WARNING && ae->new_status != RRDCALC_STATUS_CRITICAL))
-            return true;
-        ae = ae->next;
-    }
-
-    return false;
-}
-#endif
+#define SQL_GET_SNAPSHOT_ENTRIES                                                                                       \
+    " SELECT 0, hld.unique_id, hld.alarm_id, hl.config_hash_id, hld.updated_by_id, hld.when_key, "                     \
```

Comment:
I cannot see in the `WHERE` statement a necessity to have this `0` as first column. Why are you adding it here?

Reply:
This is needed for `health_alarm_log_populate` (needs a dummy sequence id) 

I will also fix alert_hash alias here